### DISCUSSION TITLE
TPI2: FileSystem - Duda Testing no detecta la excepción CanNotReadFileException

### DISCUSSION THREAD

**Discussion by matiasgratz** [dds-jv/foro#402] [Consultas de herramientas] [answered]
Buen día, haciendo el TPI2 me quedé trabado en el tema testing ya que no me estaría detectando la clase excepción CanNotReadFileException. **Sospecho que es un tema de imports o la posición del archivo.** Adjunto imagenes. ¿Despues de resolver eso, me quedarían hacer los últimos tests que es la parte de mocking verdad?
Adjunto imagenes.
Desde ya muchas gracias.
Matias Gratz.
Dejo el link al repo por las dudas: https://github.com/dds-utn/tpi2-sistema-de-archivos-2023-matiasgratz

![idea64_qaVGzX9XtQ](https://github.com/dds-jv/foro/assets/73258767/395cf6fa-d31f-43bd-aa97-2b0237230e90)
![idea64_zgW8dC9VvL](https://github.com/dds-jv/foro/assets/73258767/fbcca802-84f9-45f9-9528-5a6649ba692f)
![idea64_N9jgTeUM1T](https://github.com/dds-jv/foro/assets/73258767/e48a62c7-7714-4c21-9f3f-837af6a75902)

**Comment by flbulgarelli** [comment, marked as answer]
¡Hola!

Ojo, la sospecho que estás confundiendo la excepción `CanNotOpenFile` con `CanNotReadFile`. Son dos excepciones diferentes, que señalan cosas distintas (la imposibilidad de abrir vs de leer). Además por lo que veo ya contás con una, mientras que la otra aún no la desarrollaste.

Y efectivamente, para pasar esta entrega tenés que adaptar las interfaces y además testear el sistema de archivos de alto nivel.

¡Saludos!

**Reply by matiasgratz** [reply]
Muchas gracias profe.
